llvm/lib/Target/AMDGPU/SIShrinkInstructions.cpp

```diff
@@ -973,11 +973,13 @@ bool SIShrinkInstructions::run(MachineFunction &MF) {
         continue;
       }
 
-      if (!TII->hasVALU32BitEncoding(MI.getOpcode())) {
-        // If there is no chance we will shrink it and use VCC as sdst to get
-        // a 32 bit form try to replace dead sdst with NULL.
+      // If there is no chance we will shrink it and use VCC as sdst to get
+      // a 32 bit form try to replace dead sdst with NULL.
```

Comment:
Comment needs updating. We are now trying to replace dead sdst with null even if there _is_ a chance that we could shrink to a 32-bit encoding.